File: web/src/api/dict.ts
import request, { type PageResult } from './request'

export interface DictType {
  id: number
  dict_name: string
  dict_type: string
  status: number
  remark: string
  created_at: string
  updated_at: string
}

export interface DictData {
  id: number
  dict_type: string
  dict_label: string
  dict_value: string
  css_class: string
  list_class: string
  is_default: number
  sort: number
  status: number
  remark: string
  created_at: string
  updated_at: string
}

export interface DictTypeParams {
  page?: number
  page_size?: number
  dict_name?: string
  dict_type?: string
  status?: number
}

export interface DictDataParams {
  page?: number
  page_size?: number
  dict_type?: string
  dict_label?: string
  status?: number
}

const dictApi = {
  listTypes(params?: DictTypeParams) {
    return request.get<any, { code: number; data: PageResult<DictType> }>('/dict/types', { params })
  },

  getType(id: number) {
    return request.get<any, { code: number; data: DictType }>(`/dict/types/${id}`)
  },

  createType(data: Partial<DictType>) {
    return request.post('/dict/types', data)
  },

  updateType(id: number, data: Partial<DictType>) {
    return request.put(`/dict/types/${id}`, data)
  },

  deleteType(id: number) {
    return request.delete(`/dict/types/${id}`)
  },

  listData(params?: DictDataParams) {
    return request.get<any, { code: number; data: PageResult<DictData> }>('/dict/data', { params })
  },

  getDataByType(dictType: string) {
    return request.get<any, { code: number; data: DictData[] }>(`/dict/data/${dictType}`)
  },

  createData(data: Partial<DictData>) {
    return request.post('/dict/data', data)
  },

  updateData(id: number, data: Partial<DictData>) {
    return request.put(`/dict/data/${id}`, data)
  },

  deleteData(id: number) {
    return request.delete(`/dict/data/${id}`)
  },
}

export default dictApi

File: backend/internal/model/dict.go
package model

type DictType struct {
	SnowflakeModel
	DictName string `json:"dict_name" gorm:"size:100;not null"`
	DictType string `json:"dict_type" gorm:"size:100;not null;uniqueIndex"`
	Status   int    `json:"status" gorm:"default:1"`
	Remark   string `json:"remark" gorm:"size:255"`
}

func (DictType) TableName() string {
	return "dict_types"
}

type DictData struct {
	SnowflakeModel
	DictType  string `json:"dict_type" gorm:"size:100;not null;index"`
	DictLabel string `json:"dict_label" gorm:"size:100;not null"`
	DictValue string `json:"dict_value" gorm:"size:100;not null"`
	CssClass  string `json:"css_class" gorm:"size:100"`
	ListClass string `json:"list_class" gorm:"size:100"`
	IsDefault int    `json:"is_default" gorm:"default:0"`
	Sort      int    `json:"sort" gorm:"default:0"`
	Status    int    `json:"status" gorm:"default:1"`
	Remark    string `json:"remark" gorm:"size:255"`
}

func (DictData) TableName() string {
	return "dict_data"
}
